### ISSUE TITLE
IndexError when reading empty MultiLineString

### ISSUE THREAD

**Issue by alessioarena** [Toblerity/Shapely#935]
## Expected behavior and actual behavior.

Similar to #837, shapely fails to load MultiLineString objects with empty geometries
This in fact raises an IndexError when trying to access the first set of coordinates:

## Steps to reproduce the problem.

This problem can be reproduced by loading an empty geometry, for example:

```
import geopandas as gpd
features = [{'type': 'Feature', 'id': '21', 'properties': {'valid':'true'}, 'geometry': {'type': 'MultiLineString', 'coordinates': [[[1.0, 2.0], [1.0, 3.0]]]}}, {'type': 'Feature', 'id': '22','properties': {'valid':'false'},  'geometry': {'type': 'MultiLineString', 'coordinates': [[]]}}]
gpd.GeoDataFrame.from_features(features[:1]) # this will pass
gpd.GeoDataFrame.from_features(features[1:]) # this will fail
```

The problem sits in these lines https://github.com/Toblerity/Shapely/blob/e4afcf5d0b497c407ba8a1115e423628cbbfe337/shapely/geometry/multilinestring.py#L121-L129

In particular there is no effective test for empty geometries

## Operating system
Windows 10

## Shapely version and provenance
'1.6.4.post1' installed using conda

**Comment by sgillies** [comment]
@alessioarena we aren't supporting versions < 1.7 anymore. Can you reproduce this with version 1.7.0?

**Comment by alessioarena** [comment]
yes

```
import Python 3.6.10 | packaged by conda-forge | (default, Apr 24 2020, 16:44:11)
[GCC 7.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import shapely
>>> shapely.__version__
'1.7.0'
>>> import geopandas as gpd
>>> gpd.__version__
'0.8.0'
>>> features = [{'type': 'Feature', 'id': '21', 'properties': {'valid':'true'}, 'geometry': {'type': 'MultiLineString', 'coordinates': [[[1.0, 2.0], [1.0, 3.0]]]}}, {'type': 'Feature', 'id': '22','properties': {'valid':'false'},  'geometry': {'type': 'MultiLineString', 'coordinates': [[]]}}]
>>> gpd.GeoDataFrame.from_features(features[:1]) # this will pass
                                            geometry valid
0  MULTILINESTRING ((1.00000 2.00000, 1.00000 3.0...  true
>>> gpd.GeoDataFrame.from_features(features[1:]) # this will fail
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/flush5/are014/anaconda/envs/pyAODN/lib/python3.6/site-packages/geopandas/geodataframe.py", line 405, in from_features
    "geometry": shape(feature["geometry"]) if feature["geometry"] else None
  File "/flush5/are014/anaconda/envs/pyAODN/lib/python3.6/site-packages/shapely/geometry/geo.py", line 42, in shape
    return MultiLineString(ob["coordinates"])
  File "/flush5/are014/anaconda/envs/pyAODN/lib/python3.6/site-packages/shapely/geometry/multilinestring.py", line 52, in __init__
    self._geom, self._ndim = geos_multilinestring_from_py(lines)
  File "/flush5/are014/anaconda/envs/pyAODN/lib/python3.6/site-packages/shapely/geometry/multilinestring.py", line 123, in geos_multilinestring_from_py
    N = len(exemplar[0])
IndexError: list index out of range
>>>
```

**Comment by tomplex** [comment]
I can't reproduce this on the `maint-1.7` branch. It was solved post-1.7 release by #852. I added some test cases in #949 to prove it.

**Comment by sgillies** [comment]
Thanks @tomplex !

**Comment by sgillies** [comment]
Resolved by #935.
